StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp9/ReadabilityRules/SA1110CSharp9UnitTests.cs

```diff
@@ -3,9 +3,66 @@
 
 namespace StyleCop.Analyzers.Test.CSharp9.ReadabilityRules
 {
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Testing;
     using StyleCop.Analyzers.Test.CSharp8.ReadabilityRules;
+    using StyleCop.Analyzers.Test.Helpers;
+    using Xunit;
+    using static StyleCop.Analyzers.Test.Verifiers.StyleCopCodeFixVerifier<
+        StyleCop.Analyzers.ReadabilityRules.SA1110OpeningParenthesisMustBeOnDeclarationLine,
+        StyleCop.Analyzers.SpacingRules.TokenSpacingCodeFixProvider>;
 
     public partial class SA1110CSharp9UnitTests : SA1110CSharp8UnitTests
     {
+        [Theory]
+        [MemberData(nameof(CommonMemberData.TypeKeywordsWhichSupportPrimaryConstructors), MemberType = typeof(CommonMemberData))]
+        [WorkItem(3784, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3784")]
+        public async Task TestPrimaryConstructorWithoutParametersAsync(string typeKeyword)
+        {
+            var testCode = $@"
+{typeKeyword} Foo
+    {{|#0:(|}})
+{{
+}}";
+
+            var fixedCode = $@"
+{typeKeyword} Foo()
+{{
+}}";
+
+            var expected = this.GetExpectedResultTestPrimaryConstructor();
+            await VerifyCSharpFixAsync(testCode, expected, fixedCode, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Theory]
+        [MemberData(nameof(CommonMemberData.TypeKeywordsWhichSupportPrimaryConstructors), MemberType = typeof(CommonMemberData))]
+        [WorkItem(3784, "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3784")]
+        public async Task TestPrimaryConstructorWithParametersAsync(string typeKeyword)
+        {
+            var testCode = $@"
+{typeKeyword} Foo
+    {{|#0:(|}}int x)
+{{
+}}";
+
+            var fixedCode = $@"
+{typeKeyword} Foo(
+    int x)
```

Comment:
💡 Can you also add tests for inheritance cases?

Reply:
Just record declarations that inherit from another record? Could you describe some more what you are after?

Reply:
Yes, reference type records and classes with primary constructors allow a base type to be specified, and arguments can be provided to the base class.

```cs
record BaseRecord(int A);

record DerivedRecord(int A, int B) : BaseRecord(A);
```

```cs
class BaseClass(int a);

class DerivedClass(int a, int b) : BaseClass(int a);
```

Reply:
So the tests that you wanted was regarding the placement of the opening parenthesis **in the call to the base constructor**?

Reply:
Yes, that's correct. I believe SA1110 handles this for the equivalent `base(...)` call in constructors.

Reply:
Fixed